pkg/chart/metadata_test.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package chart
+
+import (
+	"testing"
+)
+
+func TestValidate(t *testing.T) {
+	tests := []struct {
+		md  *Metadata
+		err error
+	}{
+		{
+			nil,
+			ValidationError("chart.metadata is required"),
+		},
+		{
+			&Metadata{Name: "test"},
+			ValidationError("chart.metadata.apiVersion is required"),
+		},
+		{
+			&Metadata{APIVersion: "v2"},
```

Comment:
in this particular case, the error *may* crop up as either the name OR the version being required, because both fields are missing. Can we add a `Version` field here as well so that we can explicitly check for the name error as being done on lines 40 and 41?

Reply:
According to the code in the Validate in the metadata.go, the sequence is checked. If the APIVersion is empty, an error will be returned directly.
So I think there is no need to increase the Version field.
```
	if md == nil {
		return ValidationError("chart.metadata is required")
	}
	if md.APIVersion == "" {
		return ValidationError("chart.metadata.apiVersion is required")
	}
	if md.Name == "" {
		return ValidationError("chart.metadata.name is required")
	}
	if md.Version == "" {
		return ValidationError("chart.metadata.version is required")
	}
	if !isValidChartType(md.Type) {
```

Reply:
That’s how the code is written, yes. However, we could switch the ordering of the validation and make Helm return the version check error first, and behaviourally it would still be correct.

That’s why I’m asking to add a version to the chart metadata for this test.

Reply:
got it, updated, PTAL.